python-avd/pyavd/_schema/models/avd_model.py

```diff
@@ -79,19 +84,34 @@ def _from_dict(cls: type[T_AvdModel], data: Mapping, keep_extra_keys: bool = Fal
                 msg = f"Invalid key '{key}'. Not available on '{cls.__name__}'."
                 raise KeyError(msg)
 
-            cls_args[field] = coerce_type(data[key], cls._fields[field]["type"])
+            # Handle Dynamic Keys slightly differently
+            child_data_source = data_source if cls.__name__.startswith("Dynamic") and key == "value" else data_source.create_descendant(key)
+
+            cls_args[field] = coerce_type(data[key], cls._fields[field]["type"], data_source=child_data_source)
+            cls._field_source[field] = child_data_source
 
         if custom_data:
             cls_args["_custom_data"] = custom_data
 
-        return cls(**cls_args)
+        cls_instance = cls(**cls_args)
+        cls_instance._source = data_source
+        return cls_instance
 
     @classmethod
     def _get_field_name(cls, key: str) -> str | None:
         """Returns the field name for the given key. Returns None if the key is not matching a valid field."""
         field_name = cls._key_to_field_map.get(key, key)
         return field_name if field_name in cls._fields else None
 
+    def get_field_source(self, key: str) -> str:
```

Comment:
Here you are not prefixing with underscore like we do for all the other methods. This could clash with a model field (if we ever had one with get_ :)

Reply:
yeah will change - it is just that tehn ruff if sad when we use the function outside of the class :)